<html>

<head>
<link href="https://fonts.googleapis.com/css?family=Raleway" rel="stylesheet">
<link rel="stylesheet" href="workshop.css">
<title>ANU Nanopore Sequencing Workshop 2017</title>
</head>

<body>

<!-- ---------------------------------------------------------------- -->
<h1>ANU Nanopore Sequencing Workshop<br/>
June 14-16, 2017
</h1>

<p>
<img src='images/head_image.jpg' alt='header image'>
<h2>Abstract</h2>
Our nanopore sequencing workshop is targeted at beginners and intermediate users of </br>
the Oxford Nanopore MinION sequencing platform. It represents a unique opportunity </br>
to learn more about this exciting and mature sequencing platform from experienced users. </br>
It is an hands-on workshop run over 3 days, covering the technology, </br>
DNA sample QC, library preparation techniques (rapid, 1D and 1D^2), starting runs,</br>
bioinformatics analysis and interpretation. </br>
</br>
The workshop is based on previous efforts from leaders in the field who started <a href="https://porecamp.github.io/">UK PoreCamp</a></br>
(Josh Quick and Nick Lohman) and participated in <a href="https://porecamp-au.github.io/">porecamp.AU</a>. The workshop is targeted at</br>
the sequencing community at ANU, CSIRO, University Canberra, Australian and afar.</br>
</br>
Applications for the Nanopore sequencing workshop are now open! 10 spots are reserved</br>
for local researchers at ANU while another 20 are open to the wider community. All </br>
applications are strictly competitive and aimed at ensuring diversity.</br>
</br>
The workshop will be held at the <a pref="http://biology.anu.edu.au/research/facilities/ecogenomics-and-bioinformatics-lab-ebl">Ecogenomics and Bioinformatics Lab</a> (EBL) at the</br>
Australian National University, Canberra, from 14th to 16th of June 2017.</br>
</br>
Applications are now open and will close 1st of May 2017.
</br>
</br>
Follow us on Twitter with <a href="https://twitter.com/search?f=tweets&q=%22%23MinIONANU%22&src=typd">#MinIONANU</a>
</p>

<p>
<h3>Thanks to all who attended and made the workshop a success! We will continue to update this website with additional links and resources as they come through.</h3>
</p>

<!------------------------------------------------------------------- -->
<h2>Quick References</h2>
<ul>
<li><a href='https://www.dropbox.com/l/scl/AAADWTLuKuWqJbciF53omFrZN-qaGhX6RzE'>Dropbox containing presentation slides and notes</a>
<!-- <li><a href=''>Bioinformatic resources</a> -->

</ul>

<ul>
<li><a href='anu-nanopore.github.io/Nanopore_workshop_schedule.pdf'>Workshop schedule PDF</a>
<li><a href='https://www.dropbox.com/sh/l47xxitd6eo3dsy/AAAK_CMf1trEiASVo71JLIyqa?dl=0'>Links to MinKNOW control software</a>
<li><a href='http://www.7-zip.org/download.html'>7-zip for windows users</a>
<li><a href='https://filezilla-project.org/'>Filezilla FTP client</a>
<li><a href="https://nanoporetech.com/how-it-works">Oxford Nanopore - How nanopore sequencing works</a>
<li><a href="anu-nanopore.github.io/protocols/OXF-639 Rapid Sequencing of gDNA using SQK-RAD002_V5.pdf">Library protocol for Rapid Kit</a>
<li><a href="anu-nanopore.github.io/protocols/1D_gDNA_sequencing_SQK-LSK108_V5.pdf">Library protocol for 1D ligation Kit</a>
<li><a href="anu-nanopore.github.io/protocols/Lab_protocol_1D_2_sequencing_genomic_DNA_using_SQK-LSK308_v5.pdf">Library protocol for 1D^2 Kit</a>
</ul>

<h2>Software Links</h2>

<h3>DNA extraction:</h3>
<ul>
<li>Sambrook and Russell
<li><a href='http://lab.loman.net/'>http://lab.loman.net/</a>
<li><a href='https://docs.google.com/document/d/1025Tu_zPWqK14JEJ9IFzRul1l5RoIORE_FcAUrQpMwA/edit'>https://docs.google.com/document/d/1025Tu_zPWqK14JEJ9IFzRul1l5RoIORE_FcAUrQpMwA/edit</a> <- would benefit from a doi
<li><a href='https://www.protocols.io/groups/minion-user-group-with-fungi-and-plants-on-their-mind'>https://www.protocols.io/groups/minion-user-group-with-fungi-and-plants-on-their-mind</a>
</ul>

<h3>DNA sequencing:</h3>
<ul>
<li><a href='https://nanoporetech.com/'>nanoporetech.com</a>
<li><a href='https://docs.google.com/document/d/1025Tu_zPWqK14JEJ9IFzRul1l5RoIORE_FcAUrQpMwA/edit'>https://docs.google.com/document/d/1025Tu_zPWqK14JEJ9IFzRul1l5RoIORE_FcAUrQpMwA/edit</a>
</ul>

<h3>Looking at Fast5 files:</h3>
<ul>
<li>TBD
</ul>

<h3>Basecalling:</h3>
<ul>
<li><a href=''>Albacore</a>
<li><a href='https://github.com/nanoporetech/nanonet'>Nanonet</a>
<li><a href='https://bitbucket.org/vboza/deepnano'>DeepNano</a>
<li><a href='https://github.com/mateidavid/nanocall'>NanoCall</a>
</ul>

<h3>Post-bascalling fast5:</h3>
<ul>
<li><a href='https://github.com/arq5x/poretools'>Poretools</a>
<li><a href='https://github.com/rrwick/Porechop'>Porechop</a>
<li><a href='https://github.com/scottgigante/picopore'>Picopore</a>
<li><a href='https://github.com/mw55309/poRe_docs'>poRe</a>
<li><a href='https://documentation.tgac.ac.uk/display/NANOOK/'>NanoOK</a>
<li><a href='https://github.com/akloster/porekit-python'>Porekit</a>		 - has anyone used this?
</ul>

<h3>Mapping:</h3>
<ul>
<li><a href='https://github.com/isovic/graphmap'>GraphMap</a>
<li><a href='https://github.com/lh3/bwa'>Bwa</a>
<li><a href='http://last.cbrc.jp/'>LAST</a>
<li><a href='https://github.com/lh3/minimap'>Minimap</a>
<li><a href='https://github.com/philres/ngmlr'>NGMLR</a>
<li><a href='https://github.com/benedictpaten/marginAlign'>marginAlign</a>
<li><a href='https://github.com/ruhulsbu/NanoBLASTer'>NanoBLASTer</a>
</ul>

<h3>Assembly:</h3>
<ul>
<li><a href='https://github.com/rrwick/Unicycler'>Unicycler </a>
<li><a href='https://github.com/marbl/canu'>Canu</a>
<li><a href='https://github.com/nanoporetech/ont-assembly-polish'>Ont-assembly-polish</a>
<li><a href='https://github.com/lh3/miniasm'>Miniasm</a>
</ul>

<h3>Error correction:</h3>
<ul>
<li><a href='https://github.com/jts/nanopolish'>Nanopolish</a>
<li><a href='https://github.com/isovic/racon'>Racon</a>
</ul>

<h3>Training basecallers:</h3>
<ul>
</ul>

<h3>Real time analysis:</h3>
<ul>
<li><a href='http://minotour.nottingham.ac.uk/index.php'>MinoTour</a>
<li><a href='https://github.com/mattloose/RUscripts'>ReadUntil</a> - this doesnâ€™t really work anymore. Does it?
</ul>

<h3>Species identification:</h3>
<ul>
<li><a href='http://blast.ncbi.nlm.nih.gov/'>BLAST</a>
<li><a href='https://github.com/marbl/mash'>Mash</a>
<li><a href=''>WIMP</a>
<li><a href='https://app.onecodex.com/'>OneCodex</a>
</ul>

<h3>Modification calling:</h3>

<h3>Experimental code resources:</h3>
<ul>
<li><a href='https://github.com/rrwick/Nanopore-read-processor'>Nanopore-read-processor</a>
<li><a href='https://github.com/alexiswl/PoreCampAU'>PoreCampAU</a>
</ul>

<!------------------------------------------------------------------- -->
<h2>Workshop Details</h2>
<ul>
<li>Registration cost: $900 AUD
<li>The workshop will be held on the ANU Campus - EcoGenomics and Bioinformatics Lab</a>
<li>Workshop over June 14-16, 2017:
	<ul>
	<li>Day 1: Background, DNA sample QC, library prep, starting your run
		<li>Location <a href="http://biology.anu.edu.au/research/facilities/ecogenomics-and-bioinformatics-lab-ebl">EBL</a>
	<li>Day 2: Bioinformatic background and initial analysis of your run
		<li>Location <a href="http://www.anu.edu.au/maps#show=29145">Building 136</a>
	<li>Day 3: Bringing it all together and finishing up
		<li>Location <a href="http://www.anu.edu.au/maps#show=29145">Building 136</a>
	</ul>

</ul>
<!-- ---------------------------------------------------------------- -->
<h2>Workshop Objectives</h2>
<ul>
<li>Learn the basics of DNA extraction and DNA QC obtained from (non-)model organisms.
<li>Develop best wet-lab practices to produce a high-quality MinION library.
<li>Hands-on experience, running your MinION on your own computer.
<li>Discover suitable data-handling methods for MinION output.
<li>Discuss latest bioinformatic methods for analysis of real-time nanopore data.
</ul>
<!-- ---------------------------------------------------------------- -->
<h2>How to Apply</h2>
<ul>
<li>THE WORKSHOP HAS NOW REACHED CAPACITY, THANKS VERY MUCH FOR YOUR INTEREST
<li>Successful participants will be notified by 5th of May.
<li>Attendance fee: $900 AUD (includes flowcell & library kit for MinION run and day-time catering).
<li>You must BYO MinION device and a <a href="https://nanoporetech.com/community/lab-it-requirements">compatible laptop</a>.
<li>Accommodation not provided - recommendations to come.
<li>For more information please contact <a href="mailto:benjamin.schwessinger@anu.edu.au">Benjamin Schwessinger</a>.
</ul>
<!-- ---------------------------------------------------------------- -->
<h2>Workshop Leaders</h2>
<ul>
<li>Benjamin Schwessinger
<li>Megan McDonald
<li>Steve Eichten
</ul>
<!-- ---------------------------------------------------------------- -->
<h2>Confirmed Trainers</h2>
<ul>
<li>Benjamin Schwessinger (ANU)
<li>Megan McDonald (ANU)
<li>Steve Eichten (ANU)
<li>Louise Judd (UniMelb)
<li>Ken McGrath (AGRF)
</ul>
<!-- ---------------------------------------------------------------- -->
<h2>Sponsors</h2>
<ul>
<li><a href="http://biology.anu.edu.au/">ANU Research School of Biology</a>
<li><a href="http://biology.anu.edu.au/research/facilities/ecogenomics-and-bioinformatics-lab-ebl">ANU EcoGenomics Lab</a>
<li><a href="http://cba.anu.edu.au/about-us">ANU Centre for Biodiversity Analysis</a>
<li><a href="http://biology.anu.edu.au/centre-genomics-metabolomics-and-bioinformatics">The Centre for Genomics, Metabolomics and Bioinformatics</a>
<li><a href="http://www.plantenergy.uwa.edu.au/">ARC Centre of Excellence in Plant Energy Biology</a>
<li><a href="http://www.interpath.com.au">Interpath Services</a>
<li><a href="https://nanoporetech.com/">Oxford Nanopore</a>
</ul>
<img src='./images/image001.png'/>


</body>


</html>
